src/client/mod.rs

```diff
@@ -57,13 +60,39 @@ where
     /// ```rust
     /// # use warpgrapher::Client;
     ///
-    /// let mut client = Client::<(), ()>::new_with_http("http://localhost:5000/graphql");
+    /// let mut client = Client::<(), ()>::new_with_http("http://localhost:5000/graphql", None);
     /// ```
-    pub fn new_with_http(endpoint: &str) -> Client<(), ()> {
+    pub fn new_with_http(endpoint: &str, headers_opt: Option<HashMap<&str,&str>>) -> Client<(), ()> {
         trace!("Client::new_with_http called -- endpoint: {}", endpoint);
 
         Client::Http {
             endpoint: endpoint.to_string(),
+            headers: {
+                let mut header_map = HeaderMap::new();
+                if let Some(headers) = headers_opt {
+                    // The solution below is not elegant, but may be the best option. A header
+                    // requires a HeaderName and HeaderValue. No matter what type you convert from
+                    // you get a Result type. For example, from_str will produce an error if the
+                    // input contains "\n" because new lines are not allowed. The code below logs
+                    // any errors that are encountered, but skips any erroneous headers in creating
+                    // the header map. I imagine we don't want the Client::Http constructor to 
+                    // return a Result type to propagate errors up. 
```

Comment:
Actually `MyType::new() -> Result<MyType, Error>` is a reasonably common pattern in Rust. Fallible constructors are a thing.  I'd go that route rather than masking the error.

Reply:
I updated the constructor to return a result type and used netcat to test the ability to properly parse/send headers. No tests have been added to tests in mod.rs.